src/plugins/data_source_management/public/components/data_source_menu/data_source_menu.tsx

```diff
@@ -15,15 +15,17 @@ import {
 import { MountPointPortal } from '../../../../opensearch_dashboards_react/public';
 import { DataSourceSelectable } from './data_source_selectable';
 import { DataSourceOption } from '../data_source_selector/data_source_selector';
+import { DataSourceView } from '../data_source_view';
 
 export interface DataSourceMenuProps {
-  showDataSourceSelectable: boolean;
+  showDataSourceSelectable?: boolean;
+  showDataSourceView?: boolean;
   appName: string;
-  savedObjects: SavedObjectsClientContract;
-  notifications: NotificationsStart;
+  savedObjects?: SavedObjectsClientContract;
+  notifications?: NotificationsStart;
   fullWidth: boolean;
-  hideLocalCluster: boolean;
-  dataSourceCallBackFunc: (dataSource: DataSourceOption) => void;
+  hideLocalCluster?: boolean;
+  dataSourceCallBackFunc?: (dataSource: DataSourceOption) => void;
```

Comment:
Any reasons we set these props as optional?

Reply:
the callback function doesn't make sense for read only component, since there is no change to the selection, but only makes sense for selectables, hope it makes sense

Reply:
Thanks for the response. In that case I assume we don't need this line?:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6125/files#diff-ca572b8e5224968b4b5cfd2a810b1280195b20f6eff17b93657fefa89bb50ed1R91